### DISCUSSION TITLE
[nx-flutter] Support Libraries

### DISCUSSION THREAD

**Discussion by ghost** [tinesoft/nxrocks#24] [Ideas]
Hi, I don't know nx very well yet, but I'm missing a way to create libraries (from modules/plugins/packages), a `nx-flutter:lib` would be nice, that would help for the dependency graph, also it feels weird to call nx-flutter:app when we want to create a library (`"projectType": "library"`).
also, a library wouldn't have the `buildApk` action for example
what do you think ?
thanks

**Comment by ghost** [comment]
also a lot of options are mutually exclusive, like why choosing kotlin and swift before choosing the target as a pure Dart lib ?

some options just create errors like, a pure dart package shouldn't ask for targets, so when you choose one or more you get this : 
```
The "--platforms" argument is not supported in package template.
    Failed to execute command: flutter create --project-name=lib --org=maas.lib --description="MaaS Lib" --android-language=kotlin --ios-language=swift --template=package --platforms="android,ios,web"  apps/lib

```

**Comment by tinesoft** [comment]
Hi @benoitskipr 

Thanks for the continuing feedback, much appreciated 👍🏾 

> `a nx-flutter:lib would be nice, that would help for the dependency graph, also it feels weird to call nx-flutter:app when we want to create a library ("projectType": "library").`

Me: Good point 👍🏾 

I could add an `create` alias to the current `app` or `application`. It makes indeed more sense, and would be more aligned with the underlying `flutter create` command.  I've created #27

> also a lot of options are mutually exclusive, like why choosing kotlin and swift before choosing the target as a pure Dart lib ?
> 
> some options just create errors like, a pure dart package shouldn't ask for targets, so when you choose one or more you get this :
> 
> ```
> The "--platforms" argument is not supported in package template.
>     Failed to execute command: flutter create --project-name=lib --org=maas.lib --description="MaaS Lib" --android-language=kotlin --ios-language=swift --template=package --platforms="android,ios,web"  apps/lib
> ```

Me: Another valid point 👍🏾  

It is due to the  **static** nature of the generation prompt: The next questions don't consider the previous answers.

I will look into a way to provide a dynamic prompt (if that is possible with the **Schematics API**...). I've created  #26 for that

**Comment by ghost** [comment]
awesome, thanks for the plugin
I'm a total noob in js, coming from mobile, but would love to try
is there a tutorial on how to setup workspaces for plugin development/debug ?

**Reply by tinesoft** [reply]
Hi @benoitskipr ,

Sorry for the late answer, somehow I wasn't notified of new messages in this discussion....

Quick updates on that thread:
- the dynamic prompt now works since https://github.com/tinesoft/nxrocks/releases/tag/nx-flutter%2Fv1.3.0
- the library support has landed since version https://github.com/tinesoft/nxrocks/releases/tag/nx-flutter%2Fv2.0.0: based on the `template` type, the created project is automatically placed under the `apps` or `libs`folder in the workspace.

As for your last question, I think google is your friend! 
If you are looking for some real life projects using the plugin, maybe you an have a look at https://github.com/tinesoft/nxrocks/network/dependents?package_id=UGFja2FnZS0xNzUzMTI4ODcz, and get inspired? ;-)
